scripts/manual_validation_data_processing.R

```diff
@@ -0,0 +1,338 @@
+euctr_data <- read.csv(here::here("data", "euctr_dump.csv"), sep = ",")
+
+# Check that all trn1 from joined_data are in euctr_data
+missing_trn1 <- setdiff(joined_data$trn1, euctr_data$eudract_number)
+
+if (length(missing_trn1) == 0) {
+  cat("All trn1 values are included in eudract_number.\n")
+} else {
+  cat("The following trn1 values are missing from eudract_number:\n")
+  print(missing_trn1)
+}
+
+# Filter variables and rows of interest from euctr_data
+euctr_data_filtered <- euctr_data |>
+  select(eudract_number_with_country, eudract_number, date_of_the_global_end_of_the_trial) |>
+  filter(eudract_number %in% joined_data$trn1, 
+         str_detect(eudract_number_with_country, "DE"))
+
+# Convert dates to Date format with lubridate()
+euctr_data_filtered$date_of_the_global_end_of_the_trial <- ymd(euctr_data_filtered$date_of_the_global_end_of_the_trial)
+
+# Add completion date from EUCTR DE protocol (date_of_the_global_end_of_the_trial) to joined_data
+joined_data <- joined_data |>
+  left_join(euctr_data_filtered |>
+              select(eudract_number, 
+                     completion_date_protocol = date_of_the_global_end_of_the_trial),
+            by = c("trn1" = "eudract_number"))
+
+# Let's explore missing completion dates from the EUCTR dump for this discrepancy analysis
+# The discrepancy analysis of completion date across registrations of the same trial is limited
+# to manually validated cross-registrations where both registrations are "Completed"
+
+discrepancy_data <- joined_data |>
+  filter(
+    is_true_crossreg == TRUE,
+    overall_recruitment_status_reg1 == "Completed" & overall_recruitment_status_reg2 == "Completed"
+  )
+
+# There are 200 observations in the discrepancy analysis of completion date
+# This includes 28 observations with a missing completion_date_protocol from the EUCTR dump
+discrepancy_data |>
+  count(is.na(completion_date_protocol))
+
+# Let's explore missing the 28 completion_date_protocol data in the discrepancy data
+#   - 6 of these do not have a DE protocol in the EUCTR dump
+#   - 8 of these do not have a completion date on either the DE protocol or results (see `completion_date_type_reg1` == Missing)
+#   - 14 of these do not have a completion date on the DE protocol
+
+discrepancy_data |>
+  select(trn1, 
+         general_comment,
+         completion_date_reg1,
+         completion_date_type_reg1,
+         completion_date_protocol) |>
+  filter(is.na(completion_date_protocol)) |>
+  arrange(general_comment, desc(completion_date_type_reg1)) |>
+  print(n = Inf)
+
+# Round the completion_date_protocol column:
+joined_data <- joined_data |> 
+  mutate(
+    completion_month_year_protocol = lubridate::floor_date(completion_date_protocol, unit = "month"),
+  ) |> 
+  relocate(completion_month_year_protocol, .after = completion_month_year_reg1)
+
+### EXPORT FINAL DATASET
+
+#write.csv(joined_data, here::here("data", "manual_validation_processed.csv"), row.names = FALSE)
```

Comment:
@elisabascunan last comment, please update this to use `write_csv`:

`write_csv(joined_data, here::here("data", "manual_validation_processed.csv"))`